lib/Core/animation.ts

```diff
@@ -0,0 +1,33 @@
+const ANIMATION_TIMEOUT = 500;
+
+const transitionEnd = (element: Element | null) =>
+  new Promise<void>((resolve, reject) => {
+    if (!element) {
+      reject("Animation: Element is null");
+    } else {
+      const onEnd = () => {
+        element.removeEventListener("transitionend", onEnd);
+        resolve();
+      };
+      element.addEventListener("transitionend", onEnd);
+    }
+  });
+
+let timeoutID: ReturnType<typeof setTimeout>;
```

Comment:
I'm not entirely sure about the scoping rules here. Won't `timeoutID` be set to the id of the last call and which means `animateEnd` will clear only the last timeout event? So if `animationTimeout` is called multiple times, won't the earlier calls always result in a promise rejection?

Also is this used anywhere right now?

Reply:
hmm i think you are right, I'll try and create some tests for it although it may need a testing library like enzyme to properly emulate the events, This is currently only used in StoryPanel.exitStory() however I would like to see this replace every instance where we are using a timeout to perform some sort of state change whilst waiting for an animation

Reply:
> I would like to see this replace every instance where we are using a timeout to perform some sort of state change whilst waiting for an animation

That's a good idea. In that case it makes sense to document the purpose in the file header.

Reply:
yup will do